pykafka/cli/kafka_tools.py

```diff
@@ -54,11 +55,16 @@ def fetch_consumer_lag(client, topic, consumer_group):
     """
     latest_offsets = fetch_offsets(client, topic, 'latest')
     consumer = topic.get_simple_consumer(consumer_group=consumer_group,
-                                         auto_start=False, 
+                                         auto_start=False,
                                          reset_offset_on_fetch=False)
     current_offsets = consumer.fetch_offsets()
-    return {p_id: (latest_offsets[p_id].offset[0], res.offset)
-            for p_id, res in current_offsets}
+    consumer_id_dict = {}
+    for p_id, stat in current_offsets:
+        info = ast.literal_eval(stat[1])
```

Comment:
Though the offset metadata that pykafka creates looks like a python dictionary, it's actually a [JSON string](https://github.com/Parsely/pykafka/blob/ede795a90ac8d3212238a33cd6ef7e2514592436/pykafka/simpleconsumer.py#L887). Thus, the correct way to decode here is `json.loads(stat[1])`.

Also, since the `stat` tuple here is a `namedtuple`, it's preferable to reference the metadata field by name instead of subscript: `stat.metadata`.

Reply:
Thank you for your comment. I did some debug, and I found that stat.metadata is actually not JSON string. It's actually a byte, so json.loads(stat.metadata) will give me an error. I still need to decode the byte to string, and do json.loads. Like json.loads(stat.metadata.decode())